GameProject/Assets/TurnBasedAssets/Scripts/MouseController/MouseSelection.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using UnityEngine;
+
+namespace TurnBasedAssets.Scripts.MouseController
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
```

Comment:
Consider reshuffling these fields so public, private and serialized are all grouped and organised. Would make it easier on the eyes.